tests/eventemitter/azureeventgridtopic/azureeventgridtopic_test.go

```diff
@@ -0,0 +1,287 @@
+		checkMessage(t, count, client)
+		wg.Done()
+	}(t, 1, client)
+
+	KubectlApplyWithTemplate(t, data, "scaledObjectTemplate", scaledObjectTemplate)
+	wg.Wait()
+}
+
+// help function to load template data
+func getTemplateData() (templateData, []Template) {
+	base64EventGridKey := base64.StdEncoding.EncodeToString([]byte(eventGridKey))
+	return templateData{
+			TestNamespace:           namespace,
+			ClientName:              clientName,
+			CloudEventSourceName:    cloudeventSourceName,
+			ClusterName:             clusterName,
+			MonitoredDeploymentName: monitoredDeploymentName,
+			SutDeploymentName:       sutDeploymentName,
+			ScaledObjectName:        scaledObjectName,
+			EventGridEndpoint:       eventGridEndpoint,
+			EventGridKey:            base64EventGridKey,
+			TriggerAuthName:         triggerAuthName,
+			SecretName:              secretName,
+		}, []Template{
+			{Name: "secretTemplate", Config: secretTemplate},
+			{Name: "triggerAuthTemplate", Config: triggerAuthTemplate},
+			{Name: "cloudEventSourceTemplate", Config: cloudEventSourceTemplate},
+			{Name: "monitoredDeploymentTemplate", Config: monitoredDeploymentTemplate},
+			{Name: "sutDeploymentTemplate", Config: sutDeploymentTemplate},
+		}
+}
+
+func setupServiceBusTopicAndSubscription(t *testing.T) (*azservicebus.Client, *admin.Client) {
+	adminClient, err := admin.NewClientFromConnectionString(connectionString, nil)
+	assert.NoErrorf(t, err, "cannot connect to service bus namespace - %s", err)
+
+	topic, err := adminClient.GetTopic(context.Background(), topicName, nil)
+	assert.NoErrorf(t, err, "cannot get the topic - %s", err)
+
+	if topic == nil {
+		_, err = adminClient.CreateTopic(context.Background(), topicName, nil)
+		assert.NoErrorf(t, err, "cannot create the topic - %s", err)
+	}
+
+	subscription, err := adminClient.GetSubscription(context.Background(), topicName, subscriptionName, nil)
+	assert.NoErrorf(t, err, "cannot get the Subscription - %s", err)
+
+	if subscription == nil {
+		_, err = adminClient.CreateSubscription(context.Background(), topicName, subscriptionName, nil)
+		assert.NoErrorf(t, err, "cannot create the subscription - %s", err)
+	}
+
+	client, err := azservicebus.NewClientFromConnectionString(connectionString, nil)
+	assert.NoErrorf(t, err, "cannot connect to service bus namespace - %s", err)
+
+	return client, adminClient
+}
+
+func cleanupServiceBusTopic(t *testing.T, adminClient *admin.Client) {
+	t.Log("--- cleaning up ---")
+	_, err := adminClient.DeleteTopic(context.Background(), topicName, nil)
+	assert.NoErrorf(t, err, "cannot delete service bus topic - %s", err)
+}
+
+func checkMessage(t *testing.T, count int, client *azservicebus.Client) {
+	t.Log("--- waiting getMessage ---")
+	receiver, err := client.NewReceiverForSubscription(
+		topicName,
+		subscriptionName,
+		&azservicebus.ReceiverOptions{
+			ReceiveMode: azservicebus.ReceiveModePeekLock,
+		},
+	)
+	if err != nil {
+		panic(err)
```

Comment:
We should validate this condition instead of raising panic. If you want to stop the test immediately, use `require` instead of `assert`